base/gcutils.jl

```diff
@@ -25,6 +25,22 @@ finalizer(my_mutable_struct) do x
     ccall(:jl_safe_printf, Cvoid, (Cstring, Cstring), "Finalizing %s.", repr(x))
 end
 ```
+
+A finalizer can be registered at object constuction. In the following example note that
+it is not necessary to returned the newly created object because the finalizer returns
+it.
+
+# Example
+```julia
+
+mutable struct my_mutable_struct
+    bar
+    function my_mutable_struct(bar)
+        x = new(bar)
+        f(t) = @async println("Finalizing \$x.")
+        finalizer(f, x)
+    end
```

Comment:
```suggestion
mutable struct MyMutableStruct
    bar
    function MyMutableStruct(bar)
        x = new(bar)
        f(t) = @async println("Finalizing \$x.")
        finalizer(f, x)
    end
end
```
Also why `@async`?

Reply:
From the [docs](https://docs.julialang.org/en/v1.6-dev/base/base/#Base.finalizer): "`f` must not cause a task switch, which excludes most I/O operations such as println. Using the `@async` macro (to defer context switching to outside of the finalizer)..."

Reply:
Ah, I see, thanks for the reference. Code suggestion still applies though. :)